syntax = "proto2";
import 'nanopb.proto';
import "Engine/Maths/Maths.proto";
import "Engine/Graphics/RenderConsts.proto";

package usg.Events;

message RotateUV
{
	required float fRotation = 1;
	required uint32 uTexIndex = 2;
	required uint32 uMeshIndex = 3;
	required uint32 uUVCount = 4;
}

message TranslateUV
{
	required usg.Vector2f vTranslation = 1;
	required uint32 uTexIndex = 2;
	required uint32 uMeshIndex = 3;
	required uint32 uUVCount = 4;	
}

message ScaleModel
{
	required float fScale = 1;
}

message OverrideModelVectorEvent
{
	// TODO: Replace material name with hash
	required string varName = 1 [(nanopb).max_size = 32, (nanopb).fixed_length = true];
	required Vector4f vVector = 2;
}

message OverrideModelScalarEvent
{
	// TODO: Replace material name with hash
	required string varName = 1 [(nanopb).max_size = 32, (nanopb).fixed_length = true];
	required float fScalar = 2;
}


message ModifyAnimationConditionEvent
{
	required uint32 nameHash = 1;
	required bool value = 2;
}

message UpdateModelRenderMask
{
	required uint32 uRenderMask = 1;
}

message SetMeshRenderLayer
{
	required string meshName = 1 [(nanopb).max_size = 32, (nanopb).fixed_length = true];
	required RenderLayer eLayer = 2;
}